functions/src/webhooks/transcription.ts

```diff
@@ -1,57 +1,102 @@
 import * as functions from "firebase-functions"
 import { AssemblyAI } from "assemblyai"
-import { db } from "../firebase"
+import { db, Timestamp } from "../firebase"
 import { sha256 } from "js-sha256"
 
 const assembly = new AssemblyAI({
   apiKey: process.env.ASSEMBLY_API_KEY ? process.env.ASSEMBLY_API_KEY : ""
 })
 
 export const transcription = functions.https.onRequest(async (req, res) => {
-  if (
-    req.headers["X-Maple-Webhook"] &&
-    req.headers["webhook_auth_header_value"]
-  ) {
+  if (req.headers["x-maple-webhook"]) {
     if (req.body.status === "completed") {
+      // If we get a request with the right header and status, get the
+      // transcription from the assembly API.
       const transcript = await assembly.transcripts.get(req.body.transcript_id)
       if (transcript && transcript.webhook_auth) {
-        const maybeEventInDb = await db
+        // If there is a transcript and the transcript has an auth property,
+        // look for an event (aka Hearing) in the DB with a matching ID.
+        const maybeEventsInDb = await db
           .collection("events")
           .where("videoAssemblyId", "==", transcript.id)
           .get()
-        if (maybeEventInDb.docs.length) {
-          const authenticatedEventsInDb = maybeEventInDb.docs.filter(
-            async e => {
-              const hashedToken = sha256(
-                String(req.headers["webhook_auth_header_value"])
-              )
 
-              const tokenInDb = await db
-                .collection("events")
-                .doc(e.id)
-                .collection("private")
-                .doc("webhookAuth")
-                .get()
-              const tokenInDbData = tokenInDb.data()
-              if (tokenInDbData) {
-                return hashedToken === tokenInDbData.videoAssemblyWebhookToken
-              }
-              return false
+        if (maybeEventsInDb.docs.length) {
+          // If we have a match look for one that matches a hash of the token
+          // we gave Assembly. There should only be one of these but firestore
+          // gives us an array. If there is more than one member, something is
+          // wrong
+          const authenticatedEventIds = [] as string[]
+          const hashedToken = sha256(String(req.headers["x-maple-webhook"]))
+
+          for (const index in maybeEventsInDb.docs){
+            const doc = maybeEventsInDb.docs[index]
+
+            const tokenDocInDb = await db
+              .collection("events")
+              .doc(doc.id)
+              .collection("private")
+              .doc("webhookAuth")
+              .get()
+
+            const tokenDataInDb = tokenDocInDb.data()?.videoAssemblyWebhookToken
+
+            if (hashedToken === tokenDataInDb) {
+              authenticatedEventIds.push(doc.id)
             }
-          )
-          if (authenticatedEventsInDb) {
+          }
+
+          if (authenticatedEventIds.length === 1) {
```

Comment:
nit: Just for maintenance, could we add a few logs here:
* One if `maybeEventsInDb.length` === 0 (e.g. we're getting a webhook request for a transcription we don't know)
* One if `authenticatedEventIds.length` === 0 (e.g. the webhook doesn't match any event) 
* One if `authenticatedEventIds.length` > 1

None of these events *should* happen, but it'd make it easier to debug this flow if something goes wrong a few months down the road.